src/main/kotlin/no/nav/k9/los/nyoppgavestyring/domeneadaptere/k9klagetillos/K9KlageTilLosAdapterTjeneste.kt

```diff
@@ -0,0 +1,340 @@
+import kotlin.concurrent.fixedRateTimer
+import no.nav.k9.los.Configuration
+import no.nav.k9.los.domene.lager.oppgave.v2.TransactionalManager
+import no.nav.k9.los.domene.repository.BehandlingProsessEventKlageRepository
+import no.nav.k9.los.nyoppgavestyring.mottak.oppgave.OppgaveV3
+import kotlin.concurrent.thread
+
+class K9KlageTilLosAdapterTjeneste(
+    private val behandlingProsessEventKlageRepository: BehandlingProsessEventKlageRepository,
+    private val områdeRepository: OmrådeRepository,
+    private val feltdefinisjonTjeneste: FeltdefinisjonTjeneste,
+    private val oppgavetypeTjeneste: OppgavetypeTjeneste,
+    private val oppgaveV3Tjeneste: OppgaveV3Tjeneste,
+    private val config: Configuration,
+    private val transactionalManager: TransactionalManager
+) {
+
+    private val log: Logger = LoggerFactory.getLogger(K9KlageTilLosAdapterTjeneste::class.java)
+    private val TRÅDNAVN = "k9-klage-til-los"
+    private val MANUELLE_AKSJONSPUNKTER = AksjonspunktDefinisjon.values().filter { aksjonspunktDefinisjon ->
+        aksjonspunktDefinisjon.aksjonspunktType == AksjonspunktType.MANUELL
+    }.map { aksjonspunktDefinisjon -> aksjonspunktDefinisjon.kode }
+
+    private val AUTOPUNKTER = AksjonspunktDefinisjon.values().filter { aksjonspunktDefinisjon ->
+        aksjonspunktDefinisjon.aksjonspunktType == AksjonspunktType.AUTOPUNKT
+    }.map { aksjonspunktDefinisjon -> aksjonspunktDefinisjon.kode }
+
+    fun kjør(kjørSetup: Boolean = false, kjørUmiddelbart: Boolean = false) {
+        if (config.nyOppgavestyringAktivert()) {
+            when (kjørUmiddelbart) {
+                true -> spillAvUmiddelbart()
+                false -> schedulerAvspilling(kjørSetup)
+            }
+        } else log.info("Ny oppgavestyring er deaktivert")
+    }
+
+    private fun spillAvUmiddelbart() {
+        log.info("Spiller av BehandlingProsessEventer umiddelbart")
+        thread(
+            start = true,
+            isDaemon = true,
+            name = TRÅDNAVN
+        ) {
+            spillAvBehandlingProsessEventer()
+        }
+    }
+
+    private fun schedulerAvspilling(kjørSetup: Boolean) {
+        log.info("Schedulerer avspilling av BehandlingProsessEventer til å kjøre 10s fra nå, hver 24. time")
+        fixedRateTimer(
+            name = TRÅDNAVN,
+            daemon = true,
+            initialDelay = TimeUnit.SECONDS.toMillis(10),
+            period = TimeUnit.DAYS.toMillis(1)
+        ) {
+            if (kjørSetup) {
+                setup()
+            }
+            spillAvBehandlingProsessEventer()
+        }
+    }
+
+    private fun spillAvBehandlingProsessEventer() {
+        var behandlingTeller: Long = 0
+        var eventTeller: Long = 0
+        var forrigeOppgave: OppgaveV3? = null
+
+        log.info("Starter avspilling av BehandlingProsessEventer")
+        val tidKjøringStartet = System.currentTimeMillis()
+
+        val behandlingsIder = behandlingProsessEventKlageRepository.hentAlleDirtyEventIder()
```

Comment:
må ikke disse hentes med lås i tilfelle en annen allerede har behandlet de?

Reply:
så lenge dirty=true så vil den bli plukket på nytt?

Reply:
Tror jeg skal sove på denne, men akkurat nå tenker jeg at SKIP LOCKED må inn, for å ikke få meldinger inn i oppgavemodellen i feil rekkefølge :thinking: